tempesta_db/core/htrie.c

```diff
@@ -931,3 +931,90 @@ tdb_htrie_exit(TdbHdr *dbh)
 {
 	free_percpu(dbh->pcpu);
 }
+
+static int
+tdb_bucket_walk(TdbHdr *dbh, TdbBucket *b, int (*fn)(void *))
```

Comment:
No, these functions have different return values

Reply:
Does this really makes the unification impossible?

Reply:
Рossible through macros

Reply:
Code hidden behind macroses make me sad. But that's a personal preference. Aside from that, there are potential performance implications of such unification. Instead of direct calls, they become indirect through a function pointer.

I haven't tested performance impact of indirect vs. direct calls, so that's not clear. But here are some results from others: https://gist.github.com/rianhunter/0be8dc116b120ad5fdd4. Be sure to read the comments too, they are important.